daemon/k8s_watcher.go

```diff
@@ -932,11 +937,61 @@ func (d *Daemon) addCiliumNetworkPolicy(obj interface{}) {
 		}
 	}
 
+	var cnpns k8s.CiliumNetworkPolicyNodeStatus
 	if err != nil {
-		log.Warningf("Unable to add CiliumNetworkPolicy %s: %s", rule.Metadata.Name, err)
+		cnpns = k8s.CiliumNetworkPolicyNodeStatus{
+			OK:          false,
+			Error:       fmt.Sprintf("%s", err),
+			LastUpdated: time.Now(),
+		}
+		log.Warningf("Unable to add CiliumNetworkPolicy %s: err: '%s'. err != nil: '%t'. a nil object: '%s'", rule.Metadata.Name, err, err != nil, nil)
 	} else {
+		cnpns = k8s.CiliumNetworkPolicyNodeStatus{
+			OK:          true,
+			Error:       "OK",
+			LastUpdated: time.Now(),
+		}
 		log.Infof("Imported CiliumNetworkPolicy %s", rule.Metadata.Name)
 	}
+
+	rule.SetPolicyStatus(nodeaddress.GetName(), cnpns)
+	_, err = crdClient.Update(rule)
```

Comment:
Move this status update into its own function in `pkg/k8s`

Reply:
Will the status only work if CRD are in use?

Reply:
>Will the status only work if CRD are in use?

You got confused because of the variable name? It works for both, but I'll change the variable name to cnpClient

```
	// crdClient is the interface for CRD and TPR
	crdClient k8s.CRDInterface
```

Reply:
Renaming will help. I'm also fine keeping it as-is as CRD will be the only interface soon and TPR will be gone. I was just wondering in general.

Reply:
Fixed